src/text-fragment-utils.js

```diff
@@ -241,9 +245,9 @@ export const processTextFragmentDirective = (textFragment) => {
           const suffixResult =
               checkSuffix(textFragment.suffix, potentialMatch, searchRange);
           if (suffixResult === CheckSuffixResult.NO_SUFFIX_MATCH) {
-            missingSuffix = true;
             break;
           } else if (suffixResult === CheckSuffixResult.SUFFIX_MATCH) {
+            missingSuffix = false;
             results.push(potentialMatch.cloneRange());
             continue;
           } else if (suffixResult === CheckSuffixResult.MISPLACED_SUFFIX) {
```

Comment:
Is it intentional that we don't set `missingSuffix` to false in the `MISPLACED_SUFFIX` case?

Reply:
I refactored the code to make it clearer. The reasoning behind it is that given the finding of a match for the start term of a fragment at position P but we don't find a match for the fragment's end term followed by its suffix term ( if any ) after P, then no match for the fragment will occur after P, even when there might be other occurrences of the fragment's start term after P.
The fix adds a variable to track matches of the end term followed by the suffix term ( if any ) after the position P. After finishing looking for end/suffix terms, if no matches were found, we stop looking for fragment matches.

Reply:
That's correct, we don't want to stop the execution there. Fortunately we won't stop early on this scenario.
We only consider stopping after searching through the remainder of the DOM for an occurrence of textEnd+suffix. We have two loops for fragment matching: the external one that tries to find prefix+start and an inner one that tries to find end+suffix after the position we found prefix+start. We don't early stop/break the inner one when we don't find and occurrence of suffix after the end term, in that case we would just continue searching the DOM for an occurrence of end+suffix. We break the outer loop after the inner one didn't find any matches.

In pseudo code, assuming fragment with prefix,start, end and suffix terms:

while we haven't found more than 1 match for fragment:
  find next match for prefix followed by text start
  
  if no matches : then we already found all fragment matches, return
  match_found = false
  
  while we haven't found more than 1 match for fragment:
     try to find end+suffix match
     if match found:  match_found = true
  
  if not match_found:
     stop looking for prefix+start matches because there are no end+suffix occurrences in the remainder of the DOM

Reply:
Got it, I was not keeping track of which loop was getting broken. lgtm